tests/controllers/pagamento.controller.test.ts

```diff
@@ -0,0 +1,442 @@
+          { id: "someAnotherId", pagamentoId: "someAnotherOrderId", valorTotal: 39, tipo: PagamentoTipoEnum.Recusado},
+         ];
+
+        pagamentoUseCaseMock.getAll = jest
+          .fn()
+          .mockResolvedValue(expectedResult);
+
+          await pagamentoController.get(
+            mockRequest,
+            mockResponse,
+            mockNextFunction,
+          );
+
+          expect(mockResponse.status).toHaveBeenCalledWith(
+            StatusCode.ok,
+          );
+          expect(mockResponse.json).toHaveBeenCalledWith(expectedResult);
+          expect(mockNextFunction).not.toHaveBeenCalled();
+      });
+    });
+  });
+
+  describe("Given getById method is called", () => {
+    describe("When all the data is correct and no problems are found", () => {
+      it("should search the pagamento and return it", async () => {
+          const mockRequest = {
+              params: {
+                  id: "65a2b665e3efaec673f665bd",
+              },
+          } as any;
+          const mockResponse = {
+              status: jest.fn().mockReturnThis(),
+              json: jest.fn(),
+          } as any;
+          const mockNextFunction = jest.fn();
+
+          const expectedResult = {
+              id: "someId",
+              ...mockRequest.params,
+              pedidoId: "65a2aedd591338377f5a575e",
+              tipo: PagamentoTipoEnum.Aprovado,
+              valorTotal: 36,
+          };
+
+          pagamentoUseCaseMock.getById = jest
+              .fn()
+              .mockResolvedValue(expectedResult);
+
+          await pagamentoController.getById(
+              mockRequest,
+              mockResponse,
+              mockNextFunction,
+          );
+
+          expect(mockResponse.status).toHaveBeenCalledWith(StatusCode.ok);
+          expect(mockResponse.json).toHaveBeenCalledWith(expectedResult);
+          expect(mockNextFunction).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("When the id is not sent through the params", () => {
+      it("it should return unprocessableEntity(422) response error", async () => {
+          const mockRequest = {
+              params: {},
+          } as any;
+  
+          const mockResponse = {
+              status: jest.fn().mockReturnThis(),
+              json: jest.fn(),
+          } as any;
+  
+          const mockNextFunction = jest.fn();
+  
+          await pagamentoController.getById(
+              mockRequest,
+              mockResponse,
+              mockNextFunction,
+          );
+  
+          expect(mockResponse.status).toHaveBeenCalledWith(
+              StatusCode.unprocessableEntity,
+          );
+          expect(mockResponse.json).toHaveBeenCalledWith({
+              message: "Missing identifier id",
+          });
+          expect(mockNextFunction).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("When the useCase return an empty value", () => {
+      it("it should return notFound(404) response error", async () => {

```

Comment:
Aqui também só remover o it extra:
```suggestion
      it("should return notFound(404) response error", async () => {

```